### DISCUSSION TITLE
How to disable/replace Ctrl+D hotkey to delete line?

### DISCUSSION THREAD

**Discussion by maxijabase** [icsharpcode/AvalonEdit#299] [Q&A] [answered]
I'm implementing custom hotkeys in a code editor that uses AvalonEdit and I noticed the Ctrl+D hotkey to delete lines was nowhere to be found within the code. I was wondering If I could get a helping hand on this, thanks in advance!

**Comment by siegfriedpammer** [comment]
https://github.com/icsharpcode/AvalonEdit/blob/774b38e672b3f4252f9f4e6f2f3f6707ea7fb943/ICSharpCode.AvalonEdit/AvalonEditCommands.cs#L38-L46

hope this helps!

**Reply by maxijabase** [reply]
How would I go with overriding this KeyGesture?

**Reply by maxijabase** [reply]
I took some reference from places like https://stackoverflow.com/questions/29610545/avalonedit-ctrl-i-keybinding-doesnt-work
and https://github.com/icsharpcode/AvalonEdit/discussions/288 but I didn't seem to wrap my head around overriding the default behavior. This code editor I'm maintaining has already disabled some of the stock RoutedCommands, but the frustrating thing is I don't know where, so I can take that example instead of asking for help. I wanted to mention this is AvalonEdit 6.0.1.

**Comment by maxijabase** [comment, marked as answer]
Solved by following https://stackoverflow.com/a/29613802/10177659 carefully 😅 thank you Siegfried!
